frontend/setup-runtime.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+# Copyright (c) 2020-2021 - for information on the respective copyright owner
+# see the NOTICE file and/or the repository at
+# https://github.com/hyperledger-labs/business-partner-agent
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# First argument: Path to env.js which needs to be modified
+# Second argument: Path to BPA jar file to replace env.js
+echo "Start setting runtime variables"
+
+# Overwrite frontend runtime variables with environment variables from container
```

Comment:
What happens if the properties are not set, what is the default? Also would it make sense to skip in this case, meaning wrap this into something like if [ -z ... ];

Reply:
When the properties/env variables are not set, the system will pass an empty string ("") which replaces the value of the respective identically named environment variable in env.js. Example:

MY_VARIABLE is not set in e.g. docker-compose

which then turns this
`window.env = {
  MY_VARIABLE: "__MY_VARIABLE__",
};`

into this:

`window.env = {
  MY_VARIABLE: "",
};`

inside the jar.
If MY_VARIABLE is set to e.g. "true", "false" or anything else (e.g. "asdasd") it will be replaced accordingly.
Naturally, validation of the value needs to be done in the frontend, which for boolean values is included in this PR.
An empty string or an incorrect value does not break the code, it will be turned into "undefined".

---

Skipping this would make sense to save a little bit of startup time. We probably should discuss these kind of changes beforehand or in a different draft.

Reply:
As this is something only some users need we should keep the impact on the others as minimal as possible so I prefer the option to skip. We can do this in a second pr if you like.

Reply:
Yes, I totally agree. Since this functionality is now stable and testing takes a lot of time (rebuilding Docker images on changes) we should create a separate PR and branch to improve this script.
A.k.a if no variable is set - skip all. Only perform the edit for variables that are set.